extraction/scraper_vine.py

```diff
@@ -0,0 +1,188 @@
+import sys
+sys.path.append('../')
+import pandas as pd
+from lib.columns import clean_column_names
+from lib.clean import standardize_desc_cols, strip_birth_date
+from lib.uid import gen_uid
+from lib.path import data_file_path
+from selenium import webdriver
+from selenium.webdriver.chrome.options import Options
+from bs4 import BeautifulSoup
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+
+
+def scrape(url):
+    chrome_options = Options()
+    chrome_options.add_argument("--headless")
+    chrome_options.add_argument('--disable-gpu')
+    chrome_options.add_argument('--log-level=3')
+    browser = webdriver.Chrome(chrome_options=chrome_options)
+    browser.get(url)
+    login_button = WebDriverWait(browser, 2).until(EC.presence_of_element_located((By.ID, 'lbShowAll')))
+    login_button.click()
+
+    try:
+        inmates = []
+        soup = BeautifulSoup(browser.page_source, 'lxml')
```

Comment:
If you are only interested in the page source then I think it's better to not use Selenium. Selenium is clunky to install cross-OS and is super slow.

Also, I recommend using [scrapy](https://docs.scrapy.org/en/latest/) for anything crawling-related.

Reply:
Also committing a 10 MiB binary file like Chrome.exe is a big no-no. The common way to treat binary files is to only provide a download link or a script to install but not the file itself. Also this binary file doesn't work outside of Windows. But then again, using Selenium just to get page source is like skinning a chicken with a chainsaw.

Reply:
The html for the roster tables is hidden for a majority of the jails. This scraping method was the only one that proved successful after trying a few different approaches, spacy methods included. I'll give it another try.

Reply:
I think you'll have success with Scrapy. The data does seem to be present in page source.

Reply:
Which jails have hidden roster tables?

Reply:
I also noticed that it is present, which is why I looked for alternative solutions when I received the following output from Scrapy: 

Beauregard Parish Sheriff's Office
 Jail Roster
                                Click to register anonymously to be notified
                                upon any changes in this offender's custody status.
                                             Please wait...
Filter:
Show All
A


B

C

D

E

F

G

H

I

J   

K

L

M

N

O

P

Q

R

S

T

U

V

W

X

Y

Z
 # of offenders: Last updated: 11/19/2021 8:31:13 PM Central Time
Beauregard Parish Sheriff's Office
 Jail Roster
Click to register anonymously to be notified
                                upon any changes in this offender's custody status
                                             Please wait...
Filter:
Show All


A


B

C

D

E

F

G

H

I

J   

K

L

M

N

O

P

Q

R

S

T

U

V

W

X

Y

Z
 # of offenders: Last updated: 11/19/2021 8:31:13 PM Central Time
Please wait....
Filter:
Show All
A
B
C
D
E
F
G
H
I
J
K
L
M
N
O
P
Q
R
S
T
U
V
W
X
Y
Z
# of offenders: Last updated: 11/19/2021 8:31:13 PM Central Time


This agency is providing this roster of incarcerated offenders to the public and law enforcement in the interest of public safety. This information shall not be considered, or used as, a public document, or official document, and no other publication or copying of this information is allowed without the express written consent of this agency. Any unauthorized use of this information is forbidden and subject to criminal prosecution.

It's only when I use Selenium and Google Drive, which works to stagger to request/replicate a user, that I receive the intended output.

Reply:
I should make clear that I am able to scrape the rosters with scrapy for 4/54 agencies. I only looked for alternative methods after receiving the above output for the remaining agencies.

Reply:
I can take a look at the agencies that are not working if you have already committed the code.

Reply:
I haven't yet. I'll commit the scrapy code soon for your review. Thanks.